### PR TITLE
Some extensions for fields and field functions

### PR DESCRIPTION
**Pull request by akva2**
- allow instancing Fields directly from surface/volumes
- allow instancing FieldFunctions for a pre-existing field

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams please

File: src/Utility/FieldFunctions.C [hunk 1]
```diff
@@ -286,6 +286,16 @@ FieldFunction::FieldFunction (const std::string& fileName,
 }
 
 
+FieldFunction::FieldFunction (const std::vector<Field*>& fields)
+  : FieldFuncScalarBase("", "", "",  -1)
```
Comment (kmokstad) [inline src/Utility/FieldFunctions.C:290]: Maybe equip `FieldFuncScalarBase`with a default constructor such that this explicit forwarding is not needed?

**Review by kmokstad — APPROVED** [review]

File: src/Utility/FieldFunctions.C (hunk 1 above)
Comment (akva2) [reply src/Utility/FieldFunctions.C]: sure, i also had to do it to FieldFuncHDF5.

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Pull request merged**